File: Server/src/main/core/game/bots/ArtificialSession.java
package core.game.bots;

import core.network.IoSession;

import java.nio.ByteBuffer;

/**
 * Represents an artificial networking session.
 *
 * @author Emperor
 */
public final class ArtificialSession extends IoSession {

    private static final ArtificialSession SINGLETON = new ArtificialSession();

    private ArtificialSession() {
        super(null, null);
    }

    @Override
    public String getRemoteAddress() {
        return "127.0.0.1";
    }

    @Override
    public void write(Object context, boolean instant) {

    }

    @Override
    public void queue(ByteBuffer buffer) {
    }

    @Override
    public void write() {
    }

    @Override
    public void disconnect() {
    }

    public static ArtificialSession getSingleton() {
        return SINGLETON;
    }
}

File: Server/src/main/core/game/bots/ArtificialSession.kt
package core.game.bots

import core.network.IoSession
import java.nio.ByteBuffer

/**
 * Represents an artificial networking session.
 */
class ArtificialSession private constructor() : IoSession(null, null) {

    override fun getRemoteAddress(): String {
        return "127.0.0.1"
    }

    override fun write(context: Any, instant: Boolean) {
    }

    override fun queue(buffer: ByteBuffer) {
    }

    override fun write() {
    }

    override fun disconnect() {
    }

    companion object {
        val singleton: ArtificialSession = ArtificialSession()
    }
}
